cli/tests/test_parallelize_command.rs

```diff
@@ -0,0 +1,206 @@
+// Copyright 2024 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::path::Path;
+
+use crate::common::TestEnvironment;
+
+#[test]
+fn test_parallelize_no_descendants() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
+    let workspace_path = test_env.env_root().join("repo");
+
+    for n in 1..6 {
+        test_env.jj_cmd_ok(&workspace_path, &["commit", &format!("-m{n}")]);
+    }
+    test_env.jj_cmd_ok(&workspace_path, &["describe", "-m=6"]);
+    insta::assert_snapshot!(get_log_output(&test_env, &workspace_path), @r###"
+    @  b911505e443e 6
+    ◉  2e00cb15c7b6 5
+    ◉  9df3c87db1a2 4
+    ◉  9f5b59fa4622 3
+    ◉  d826910d21fb 2
+    ◉  dc0e5d6135ce 1
+    ◉  000000000000
+    "###);
+
+    test_env.jj_cmd_ok(&workspace_path, &["parallelize", "dc0::"]);
+    insta::assert_snapshot!(get_log_output(&test_env, &workspace_path), @r###"
+    @  6c7b60a45eb6 6
+    │ ◉  296f48966777 5
+    ├─╯
+    │ ◉  524062469789 4
+    ├─╯
+    │ ◉  a9334ecaa379 3
+    ├─╯
+    │ ◉  3a7b37ebe843 2
+    ├─╯
+    │ ◉  761e67df44b7 1
+    ├─╯
+    ◉  000000000000
+    "###);
+}
+
+// Only the head commit has descendants.
```

Comment:
Can we delete this test case now or do you think it might catch regressions that the next test would not catch?

Reply:
I think we should leave this test. The first two tests are probably the simplest and most common uses for this feature. If nothing else, it's nice to have these as documentation of how it works. I can create fewer nodes if you think they're taking up too many lines of code.